### PR TITLE
fix(link): multi-cluster:start using native openshift linking service

### PR DESCRIPTION
**Pull request by sbose78**
**--- do not merge  ---**

**Depends on**

1. https://github.com/fabric8-ui/ngx-login-client/pull/98

2. The `Getting Started` page currently uses the `https://sso.prod-preview.openshift.io/auth/realms/fabric8/broker/openshift-v3/token` url to determine if the user is connected. This has to be changed to `GET /api/token?for=<user's cluster url` . 

@dgutride , I see that we have helper methods to fetch tokens in the Provider service as well as the `ngx-login-client`'s AuthenticationService . 

the method called here https://github.com/fabric8-ui/ngx-login-client/blob/master/src/app/auth/authentication.service.ts#L41 now needs the user's cluster url too.
Do we make changes in that method? Or shall we get the helper methods in provider ts ?

**What this PR accomplishes**
Previously, we linked Github using the Auth service & OpenShift using keycloak.
In this PR, we are linking OpenShift using Auth - this would be critical for multi-cluster support.

### CODE REVIEW

**Comment by sbose78** [comment]
tests pending!

**Comment by dgutride** [comment]
We need to make token retrieval all go through the ngx-login-client/authenticationservice helper method - where is the provider method you referred to @sbose78 ?

**Comment by maxandersen** [comment]
related to https://github.com/openshiftio/openshift.io/issues/1685

**Comment by sbose78** [comment]
I see we are calling `getGithubStatus()` here, https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/shared/account/provider.service.ts#L73

The URL pattern for getting the oso token is the exactly the same as above ( except of course the extra param `force_pull` ). So that makes me wonder if we were moving things here..

If we are gonna keep using the ngx-login-client, then we would need to change the way we do https://github.com/fabric8-ui/ngx-login-client/blob/master/src/app/auth/authentication.service.ts#L41

` this.openShiftToken = this.createFederatedToken(this.openshift, (response: Response) => response.json() as Token);`

needs to be something like

` this.openShiftToken = this.createFederatedToken(this.openshift, user_cluster, (response: Response) => response.json() as Token);`

**Comment by sbose78** [comment]
@dgutride I'm a bit unsure if we can pass the user's cluster url when we call `this.openshiftToken` this.openShiftToken = this.createFederatedToken(this.openshift, user_cluster, (response: Response) => response.json() as Token);

I'm a bit unsure how to do so, any pointers or design suggestions ?

File: src/app/shared/account/provider.service.ts [hunk 1]
```diff
@@ -115,8 +105,22 @@ export class ProviderService {
    * @param redirect URL to be redirected to after successful account linking
    */
   link(provider: string, redirect: string): void {
-    let url = this.getLegacyLinkingUrl(provider, redirect);
-    this.redirectToAuth(url);
+    let linkURL = this.linkUrl + '?for=' + provider + '&redirect=' +  encodeURIComponent(redirect) ;
+    console.log('attempting to connect ' + provider + ' using ' + linkURL);
```
Comment (dgutride) [inline src/app/shared/account/provider.service.ts]: console.log should be removed - there is a log service

File: src/app/shared/account/provider.service.ts [hunk 2]
```diff
@@ -115,8 +105,22 @@ export class ProviderService {
...
+    this.http
+    .get(linkURL)
+    .map(response => {
+      // TODO: what happens to this when the response is not a pure json
+      let redirectInfo = response.json() as Link;
+      this.redirectToAuth(redirectInfo.redirect_location);
+
+      // todo - handle redirect info?
+      console.log(redirectInfo);
```
Comment (dgutride) [inline src/app/shared/account/provider.service.ts]: more console.logs - pls remove or replace

File: src/app/shared/account/provider.service.ts [hunk 3]
```diff
@@ -115,8 +105,22 @@ export class ProviderService {
...
+    })
+    .catch((error) => {
+      console.log('error while linking ' + linkURL);
```
Comment (dgutride) [inline src/app/shared/account/provider.service.ts]: See above...

**Comment by joshuawilson** [comment]
[test]

**Comment by dgutride** [comment]
This should be closed now @sbose78 - moving to closed, just so it doesn't get merged.

**Comment by sbose78** [comment]
Thanks!

**Pull request closed without merging**